docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/utilities/index.js

```diff
@@ -57,3 +57,73 @@ export const getShouldLoadChartWrapper = (pathName, rules) => {
 
     return false
 };
+
+/**
+ * buildAxisData - Builds out axis data labels by running against provided formatter
+ *
+ * @param {[]} axisData
+ * @param labelFormatter
+ * @returns {*}
+ */
+export function buildAxisData(axisData, labelFormatter) {
+  for (let i = 0; i < axisData.length; i++) {
+    if (axisData[i].labels.hasFormatter) {
+      delete axisData[i].labels.hasFormatter;
+      axisData[i].labels.formatter = labelFormatter[i].formatter;
+    }
+  }
+  return axisData;
+};
+
+/**
+ * getChartData - Returns chart data based on provided data file name
+ *
+ * @param {String} dataFileName
+ * @returns {*}
+ */
+export const getChartData = (dataFileName) => {
+  const sessionKey = 'factBook';
+
+  // Get chart data from session
+  const chartData = getSessionChartData(sessionKey);
+
+  // Early return if chart session data is found for requested file name
+  if(Object.keys(chartData).length !== 0 && chartData[dataFileName]) {
+    return chartData[dataFileName];
+  }
+
+  const baseURL = window.CDEConfig.chartData.factBook.baseUrl;
+  const fileType = window.CDEConfig.chartData.factBook.dataType;
+  const fileURL = `${baseURL}/${dataFileName}.${fileType}`;
+
+  const httpReq = new XMLHttpRequest(); // a new request
+  httpReq.open("GET",fileURL,false);
```

Comment:
Is there a reason for making this a synchronous call?

The [MDN docs for the async paramter](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open#parameters) call this out as something to avoid.

Reply:
Oh snap, that wasn't intentional, was tinkering at some point and added the 3rd param. Thanks for the catch.

Reply:
Fixed.